docs/remote-bazel.md

```diff
@@ -371,3 +400,100 @@ bb remote \
 
 As the remote runners will use this SSH configuration to access your repo,
 containerization is not supported.
+
+### Troubleshooting and Common Issues
+
+#### Using an API key with Action Cache write support
+
+To take advantage of recycled runners, you must use an API key that
+supports writing to the remote Action Cache.
+
+Snapshot metadata is stored in the Action Cache. If this metadata is not successfully
+cached, future runs will not be able to use it to restore from a recycled runner.
+
+If your API key does not have AC write support, you may see an error like
+`WARNING: --remote_upload_local_results is set, but the remote cache does not
+support uploading action results or the current account is not authorized to
+write local results to the remote cache.`
+
+#### Why was my remote run unexpectedly slow?
+
+If a remote run was unexpectedly slow, one common culprit is that it didn't
+resume from a recycled snapshot.
+
+If your run started from a recycled snapshot, you should expect to see:
+
+- `Syncing existing repo...` at the top of the remote runner logs. If it
+  started from a clean runner, it would say `git init` instead.
+- `VM resumed from snapshot ID` in the execution metadata in the `Executions`
+  tab for your remote run invocation. This section contains information about the
+  snapshot key and invocation your current run resumed from.
+
+#### Why isn't my remote run using a recycled runner?
+
+If your remote runs aren't using a recycled runner when you expected them to,
+there are a couple common culprits.
+
+_Not using an API key with Action Cache write support_
+
+See [above](#using-an-api-key-with-ac-write-support) for more details.
+
+_The snapshot was evicted_
+
+Recycled runner snapshots are stored in the remote cache and are subject to eviction
+like any other cache artifacts. If the cache TTL has expired, the snapshot may
+have been evicted and may no longer be accessible.
+
+_Snapshot key mismatch_
```

Comment:
This section also applies to workflows, and is sort of already covered in the new "Remote runner introduction" doc if you wanted to link to that instead? might make it easier to keep these docs up to date, but it's also fine to inline it here

Reply:
Linked to the other doc, but kept a short blurb inlined